### ISSUE TITLE
cups-filters 2.0.0 partially break printing colored data on a B/W printer using IPP Everywhere

### ISSUE THREAD

**Issue by l3u** [OpenPrinting/libcupsfilters#52]
Dear devs,

I use a Brother HL-L2370DN B/W Duplex laser printer. I added it to CUPS using the IPP Everywhere driver, and until recently, it worked without a problem, using cups-filters-1.28.17 on Gentoo Linux.

However the recent update to cups-filters-2.0.0 broke printing when colored data is to be printed.

The color wheel from the CUPS test page looks like this:
![1](https://github.com/OpenPrinting/cups-filters/assets/6770781/cd758e1a-c6ab-4d0c-a4f7-b3e96eccc613)
And when printing some red text, it's simply omitted:
![3](https://github.com/OpenPrinting/cups-filters/assets/6770781/321c3f60-17bf-4ebd-b3cd-7bf8413dd372)

Downgrading to cups-filters-1.28.17 fixes the problem.

The relevant test page part looks like this again:
![5](https://github.com/OpenPrinting/cups-filters/assets/6770781/d27b46ab-db26-4c8f-8664-cd458d14c186)
and red text is printed again in a gray tone:
![6](https://github.com/OpenPrinting/cups-filters/assets/6770781/cc2b32e0-0ffd-4875-a1ae-598e38ccaa07)

I would really love to help tracking this down, as all the new printers use IPP Everywhere … Thanks for fixing this in advance!

Cheers, Tobias

**Comment by l3u** [comment]
Thanks for the follow-up!

Interestingly, I downgraded Gentoo's net-print/cups-filters package from 2.0.0 to 1.28.17-r2, and that made it work again. There's also a net-print/libcupsfilters package, but there's only version 2.0.0-r3 available atm, and the downgrade didn't touch it!

So if this is a libcupsfilters issue, why does downgrading cups-filters fix it?

And are those patches already released? Otherwise, I would try them …

**Comment by l3u** [comment]
Hey, thanks for your hints!

I applied all three patches (for c6175a2f, I had to remove the first hunk so that it worked, but this one only changes whitespace), rebuild libcupsfilters, installed it and restared cupsd. After that, driverless printing works as before.

Nice :-)

I'll forward this to Gentoo, maybe, they include those patches, before the next release. However, I think this would be worth a bugfix release, as this is really serious breakage – and it's already fixed upstream!
